core/corehttp/gateway_handler.go

```diff
@@ -153,12 +154,23 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	nd, err := core.Resolve(ctx, i.node, path.Path(urlPath))
-	// If node is in offline mode the error code and message should be different
-	if err == core.ErrNoNamesys && !i.node.OnlineMode() {
-		w.WriteHeader(http.StatusServiceUnavailable)
-		fmt.Fprint(w, "Could not resolve path. Node is in offline mode.")
+	switch err {
+	case nil:
+		// core.Resolve worked
+	case core.ErrNoNamesys:
+		if !i.node.OnlineMode() {
```

Comment:
What happens if we get ErrNoNamesys and we're 'online' ? Even if this can never happen we should still check that case here and log an error